bundles/org.openhab.binding.remehaheating/src/main/java/org/openhab/binding/remehaheating/internal/RemehaApiClient.java

```diff
@@ -0,0 +1,395 @@
+     * 1. Generates PKCE code verifier and challenge
+     * 2. Initiates OAuth2 authorization request
+     * 3. Extracts CSRF token from response cookies
+     * 4. Submits user credentials
+     * 5. Retrieves authorization code from redirect
+     * 6. Exchanges authorization code for access token
+     * 
+     * @param email Remeha Home account email
+     * @param password Remeha Home account password
+     * @return true if authentication successful, false otherwise
+     */
+    public boolean authenticate(String email, String password) {
+        try {
+            codeVerifier = generateRandomString();
+            String codeChallenge = generateCodeChallenge(codeVerifier);
+            String state = generateRandomString();
+
+            String authUrl = buildAuthUrl(codeChallenge, state);
+            Request authRequest = httpClient.newRequest(authUrl).method(HttpMethod.GET);
+
+            ContentResponse response = authRequest.send();
+            String requestId = response.getHeaders().get("x-request-id");
+            String csrfToken = extractCsrfToken(response);
+
+            if (csrfToken == null || requestId == null) {
+                logger.debug("Failed to extract CSRF token or request ID");
+                return false;
+            }
+
+            String stateProperties = createStateProperties(requestId);
+            if (!submitCredentials(email, password, csrfToken, stateProperties)) {
+                return false;
+            }
+
+            String authCode = getAuthorizationCode(csrfToken, stateProperties);
+            if (authCode == null) {
+                return false;
+            }
+
+            return exchangeCodeForToken(authCode);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.debug("Authentication interrupted", e);
+            return false;
+        } catch (Exception e) {
+            logger.debug("Authentication failed", e);
+            return false;
+        }
+    }
+
+    /**
+     * Retrieves the dashboard data containing all heating system information.
+     * 
+     * The dashboard includes:
+     * - Appliance information (boiler status, water pressure)
+     * - Climate zones (room temperature, target temperature)
+     * - Hot water zones (DHW temperature, mode, status)
+     * - Outdoor temperature information
+     * 
+     * @return Dashboard JSON object or null if request fails
+     */
+    public @Nullable JsonObject getDashboard() {
+        if (accessToken == null) {
+            return null;
+        }
+        try {
+            ContentResponse response = httpClient.newRequest(API_BASE_URL + "/homes/dashboard").method(HttpMethod.GET)
+                    .header("Authorization", "Bearer " + accessToken)
+                    .header("Ocp-Apim-Subscription-Key", SUBSCRIPTION_KEY).send();
+            return gson.fromJson(response.getContentAsString(), JsonObject.class);
```

Comment:
Did you consider to add logic to check if there is an authentication / oauth issue and a re-authentication is needed?